### PR TITLE
Hashmap bite

### PR DESCRIPTION
**Pull request by bbelderbos**
should have done on fork, will do next time ...

### CODE REVIEW

File: string_to_hashmap/src/main-template.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+use std::collections::HashMap;
+
+fn parse_query_params(query: &str) -> HashMap<String, String> {
+    // Implement the function to parse query parameters from the input string
+}
+
+
+#[cfg(test)]
+mod tests {
```
Comment (bbelderbos) [inline string_to_hashmap/src/main-template.rs:9]: current approach of copying main to main-template does lead to duplicate code, what if we need to update the tests, we'd need to update it in two places, maybe we update the instructions to just copy the function to the template file and not requiring the copying of the tests?

File: string_to_hashmap/src/main-template.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+use std::collections::HashMap;
+
+fn parse_query_params(query: &str) -> HashMap<String, String> {
```
Comment (bbelderbos) [inline string_to_hashmap/src/main-template.rs:3]: I want to add a bit of linting with pre-commit, might do that here or in a new PR ...

File: string_to_hashmap/exercise.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+Update the `parse_query_params` function to parse a URL query string (e.g., `"name=John&age=30"`) and return a `HashMap<String, String>` of key-value pairs.
+
+## Example:
+
+Given the input string `"name=John&age=30"`, the function should return a `HashMap` containing:
+
+- `name` -> `"John"`
+
+- `age` -> `"30"`
```
Comment (Tanner-Ray-Martin) [inline string_to_hashmap/exercise.md:17]: This makes a great reference for my future exercise descriptions. awesome!

File: string_to_hashmap/src/main-template.rs (hunk 1 above)
Comment (Tanner-Ray-Martin) [reply string_to_hashmap/src/main-template.rs:9]: I like that idea.

File: string_to_hashmap/src/main-template.rs (hunk 2 above)
Comment (Tanner-Ray-Martin) [reply string_to_hashmap/src/main-template.rs:3]: Thinking the same thing yesterday. good rust formatter? or is there one built in?

**Review by Tanner-Ray-Martin — APPROVED** [review]
awesome, wish i had this bite a few weeks ago :). tests passed and looks good!

File: string_to_hashmap/src/main-template.rs (hunk 2 above)
Comment (bbelderbos) [reply string_to_hashmap/src/main-template.rs:3]: I think so: cargo fmt :)
it's all in  pre-commit now

File: string_to_hashmap/src/main-template.rs (hunk 1 above)
Comment (bbelderbos) [reply string_to_hashmap/src/main-template.rs:9]: ok will do that update quickly on main after merging all in

**Comment by bbelderbos** [comment]
> awesome, wish i had this bite a few weeks ago :). tests passed and looks good!

yeah I think it's important the author can run the tests locally :)

**Comment by bbelderbos** [comment]
thanks for reviewing.

**Pull request merged**